lib/internal/Magento/Framework/App/HttpRequestInterface.php

```diff
@@ -0,0 +1,52 @@
+<?php
```

Comment:
Maybe you have this [class](https://github.com/magento/zf1/blob/master/library/Zend/Controller/Request/Http.php) for this purpose.